### PR TITLE
Set message priority

### PR DESCRIPTION
**Pull request by paran01d**
The pull request is for adding support for setting the message priority for the rabbitmq messge.

### CODE REVIEW

**Comment by jbarreneche** [comment]
Hi! It's been a while since we do any work around this!

As far I can see, the priority option was introduced in [RabbitMQ 3.5](http://rubybunny.info/articles/exchanges.html#message_priority), we would like to keep support for RabbitMQ 3.4 (I know we are a bit outdated!). Would you mind handling the case differently when the option is not provided (eg. when default is `null` not sending the option)?

Also, do you happen to know since which version of bunny this option is supported? (in case we need to change the minimum required version of Bunny in the gemspec)

Thanks!

**Comment by paran01d** [comment]
Yeah we have been using this plugin for a couple of years now, without wanting this feature.

Ok based on the tags from: https://github.com/ruby-amqp/bunny/commit/a2527f3c8c96068ad5d75c04f9b1459f35be47da#diff-513c0f343d97c93203c36a0a34dc93ac

Looks like priority was added in bunny 1.1.6, ill do some work on making the priority option.. errm optional. :)

File: lib/fluent/plugin/out_amqp.rb [hunk 1]
```diff
@@ -67,7 +67,11 @@ def format(tag, time, record)
   def write(chunk)
     chunk.msgpack_each do |(tag, time, record)|
       event = @payload_only ? record : { "key" => tag, "timestamp" => time, "payload" => record }
-      get_or_create_exchange.publish Yajl.dump(event), routing_key: tag, content_type: @content_type, priority: @priority
+      puboptions = { routing_key: tag, content_type: @content_type }
+      if ( !@priority.nil? )
```
Comment (jbarreneche) [inline lib/fluent/plugin/out_amqp.rb]: `if @priority`

File: lib/fluent/plugin/out_amqp.rb [hunk 2]
```diff
@@ -67,7 +67,11 @@ def format(tag, time, record)
...
+          puboptions[:priority] = @priority
```
Comment (jbarreneche) [inline lib/fluent/plugin/out_amqp.rb]: You used 4 spaces instead of 2

File: README.md [hunk 3]
```diff
@@ -43,6 +43,7 @@ Or install it yourself as:
   exchange_durable true # optionally set exchange durability - default is true.
   payload_only false # optional - default is false. if true, only the payload will be sent. if false, data format is { "key" => tag, "timestamp" => time, "payload" => record }.
   content_type application/octet-stream # optional - default is application/octet-stream. some amqp consumers will expect application/json.
+  priority 0 # the priority for the message
```
Comment (jbarreneche) [inline README.md]: Could you add the comment of the minimum `bunny` version for this option and minimum `rabbitmq`?

**Comment by jbarreneche** [comment]
@paran01d do you want me to wrap up the changes? I don't mind fixing the comments I made, just wanted to let you complete the feature. Just, let me know!

**Comment by paran01d** [comment]
Thats ok ill wrap it up today. Been busy elsewhere.

**Comment by jbarreneche** [comment]
@paran01d Thanks a lot! I've just released v0.2.0 with your changes 😄

**Pull request merged**